mlrun/model_monitoring/controller.py

```diff
@@ -443,154 +418,71 @@ def model_endpoint_process(
                 )
 
                 for start_infer_time, end_infer_time in batch_window.get_intervals():
-                    # start - TODO : delete in 1.9.0 (V1 app deprecation)
-                    try:
-                        # Get application sample data
-                        offline_response = cls._get_sample_df(
-                            feature_set=m_fs,
+                    prediction_metric = tsdb_connector.read_predictions(
+                        endpoint_id=endpoint_id,
+                        start=start_infer_time,
+                        end=end_infer_time,
+                    )
+                    if isinstance(
+                        prediction_metric,
+                        mm_schemas.ModelEndpointMonitoringMetricNoData,
+                    ):
```

Comment:
I suggest this more direct check:
```suggestion
                    if not prediction_metric.data:
```